liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/ServerFeatureSupport.java

```diff
@@ -54,6 +54,19 @@ public abstract class ServerFeatureSupport extends BasicSupport {
      */
     @Parameter( defaultValue = "${plugin}", readonly = true )
     private PluginDescriptor plugin;
+    
+    /**
+     * Packages to install. One of "all", "dependencies", "project", or "spring-boot-project".
+     */
+    @Parameter(property = "deployPackages", defaultValue = "project", alias = "installAppPackages")
```

Comment:
Is it ok to move this up two levels in the class hierarchy? It makes it apply to a lot more goals than it did previously. Not sure if that matters if we don't document it that way or use it for the other goals.

Reply:
I was hesitant to do this for that reason.... I originally had the spring boot check in the DevMojo and RunServerMojo classes which both extend from the PluginConfigSupport class where this originally was. I then passed the value down to ServerFeatureSupport.getRelevantMultiModuleProjects().  The issue came from GenerateFeaturesMojo which also has this multi-mod check. That class is one of the ones that does not extend PluginConfigSupport, so I could not get at that deployPackages property.

Reply:
So would it make more sense to change `GenerateFeaturesMojo` to extend `PluginConfigSupport` instead? Not sure what the implications would be there.

Reply:
At this point I think the inheritance type hierarchy is maybe overly prioritizing code reuse without making much sense at all from a domain-driven perspective (where we can say "B is an A" when B extends A).

Instead of shuffling things around further... maybe a better approach would be to extract the common code out into a new class. e.g a new "MultiModuleReducer" (or whatever name). 
 This could be constructed with the current project and properties and could be used commonly without messing with the hierarchy.   

Another idea would be to add more static methods to MavenProjectUtil and also avoid using parent class methods...but I like the idea of the new class better.

Reply:
I'm realizing my last comment doesn't really address adding the common config (`@Parameter`) in the PR here, while suggesting doing more work/refactoring of the logic.

I do think there's some value in being more selective with the config.. of only having the deployPackages config parm for goals for which deploy is really relevant...like deploy, run, dev (and, well, I guess create if we leave it in PluginConfigSupport but I'm not saying we have to go 100% in this direction). 

So where does that leave us for generateFeatures and the Spring Boot JAR package?   Well, at this point in the week, I'm not sure I know... but it does seem maybe more correct to have to answer this question independently, even if we need yet one more method to do so, because it  doesn't seem like it should just necessarily have the same logic applied as run or dev.

Reply:
@cherylking  - that's a good point. We could just leave the deploy packages config in PluginConfigSupport (where is belongs) and have GenerateFeaturesMojo extend it. 

@scottkurz  - If I'm reading your comments correctly, you are agreeing with that in that we should be more selective with the config and not open the deploy packages config to goals where it is not applicable. 

I will go ahead with that change.